### PR TITLE
Update Dockerfile for eth

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: scripts/docker/eth.dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# Build stage
+FROM alpine:latest as builder
+RUN apk add --no-cache \
+        git \
+        cmake \
+        g++ \
+        make \
+        linux-headers \
+        leveldb-dev --repository http://dl-cdn.alpinelinux.org/alpine/edge/testing/
+RUN git clone --recursive https://github.com/ethereum/cpp-ethereum --branch develop --single-branch --depth 1
```
Comment (chriseth) [inline scripts/docker/eth.dockerfile]: I'm also not an expert, but I think if you use multiple run-statements, the intermediate snapshots will be stored, so your image will be larger.

File: scripts/docker/eth.dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+RUN mkdir /build
+RUN cd /build && cmake /cpp-ethereum -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=RelWithDebInfo -DTOOLS=OFF -DTESTS=OFF
+RUN cd /build && make -j8 && make install
+
+# Install stage
+FROM alpine:latest
```
Comment (chriseth) [inline scripts/docker/eth.dockerfile]: I have never seen multiple from statements - which effect does that have?

Comment (chfast) [reply scripts/docker/eth.dockerfile]: I've just found it today by inspecting `--squash` option: https://docs.docker.com/engine/userguide/eng-image/multistage-build/.

In this case we have to images: builder and eth. The builder one is discarded, so multiple layers does not matter. In the "eth" stage we just copy the binary from the "builder" stage. The final image is 125 Mb.

**Comment by holiman** [comment]
Lgtm!

File: scripts/docker/eth.dockerfile (hunk 2 above)
Comment (chriseth) [reply scripts/docker/eth.dockerfile]: Oh that's nice! Still, 125 MB sounds quite big to me. Hm, ok, that's probably the debugging info:

My local eth binary is 140MB big, after stripping, it has 8.3 MB.

Comment (chfast) [reply scripts/docker/eth.dockerfile]: Yes, it includes debug symbols.

**Pull request merged**